test/api/core_test.dart

```diff
@@ -108,6 +112,156 @@ void main() {
       check(result).equals(3);
     });
   });
+
+  test('API network errors', () async {
+    Future<void> checkRequest<T extends Object>(
+        T exception, Condition<NetworkException> condition) {
+      return check(tryRequest(exception: exception))
+        .throws<NetworkException>(it()
+          ..routeName.equals(kExampleRouteName)
+          ..cause.equals(exception)
+          ..which(condition));
+    }
+
+    checkRequest(http.ClientException('Oops'), it()..message.equals('Oops'));
+    checkRequest(const TlsException('Oops'), it()..message.equals('Oops'));
+    checkRequest((foo: 'bar'), it()..message.equals('Network request failed'));
+  });
+
+  test('API 4xx errors, well formed', () async {
+    Future<void> checkRequest({
+      int httpStatus = 400,
+      String? code = 'SOME_ERROR',
+      String? expectedCode,
+      String message = 'A thing failed',
+      Map<String, dynamic> data = const {},
+    }) async {
+      final json = {
+        'result': 'error',
+        if (code != null)
+          'code': code,
```

Comment:
> with a newline/indent convention like the form here, I could imagine one could write the following but not know, or forget, that it would misbehave

Certainly worth worrying about in general — that sort of bad code has been known to become a serious problem, notably with the "goto fail" vulnerability:
https://www.imperialviolet.org/2014/02/22/applebug.html
https://dwheeler.com/essays/apple-goto-fail.html

(In that instance I doubt anyone wrote that code, looked at it, and just didn't know or remember how the syntax works. Rather it was presumably a bad merge, or bad copy-paste, and then a lack of code review and of adequate static analysis, in particular for dead code.)

For this particular construct, the upstream Flutter style guide goes for the single-line form too where possible:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#prefer-single-line-for-short-collection-if-and-collection-for

I'll go ahead and change it.